packages/wrangler/src/deployment-bundle/bundle.ts

```diff
@@ -507,6 +512,18 @@ export async function bundleWorker(
 		);
 	}
 
+	const notExportedWorkflows = workflowBindings
+		.filter((x) => !x.script_name && !entryPoint.exports.includes(x.class_name))
+		.map((x) => x.class_name);
+	if (notExportedWorkflows.length) {
+		const relativePath = path.relative(process.cwd(), entryFile);
+		throw new UserError(
+			`Your Worker depends on the following Workflows, which are not exported in your entrypoint file: ${notExportedWorkflows.join(
+				", "
+			)}.\nYou should export these objects from your entrypoint, ${relativePath}.`
```

Comment:
Perhaps worth including something like "or, if defined in another script, add `script_name` to your config"